src/pipdeptree/_util.py

```diff
@@ -0,0 +1,7 @@
+from __future__ import annotations
+
+import re
+
+
+def pep503_normalize(name: str) -> str:
+    return re.sub("[-_.]+", "-", name).lower()
```

Comment:
I'm 99% sure that the packaging library we depend on already offer this out of the box and we don't need to duplicate it here.